engine/src/main/java/org/hibernate/search/bridge/builtin/time/impl/LocalDateNumericBridge.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.bridge.builtin.time.impl;
+
+import java.time.LocalDate;
+
+/**
+ * Converts a {@link LocalDate} to a {@link Long}.
+ * <p>
+ * The digits of the resulting number will have the following pattern: yyyyMMdd
+ * <p>
+ * For example, the date 2001-3-25 will become the number 20010325.
```

Comment:
interesting POV. I had suggested the numeric encoding to Davide as it's more efficient from both space and query perspective.
When we do projections - or anything - we return the original type to the user, so this would only be problematic with low-level raw access (like you say): I'm not sure we should encourage that? 

It boils down to a need of project mission statement I think; we should decide if we're *using* the index to provide additional powerful query capabilities, or *map* to the index and allow the user to choose their own mapping.

I'd be inclined to say that we don't do mapping and that "raw index access" is at your own risk...